compiler/sem/sem.nim

```diff
@@ -368,9 +368,9 @@ func getDefNameSymOrRecover*(n: PNode): PSym {.inline.} =
   of nkError:
     case n.diag.kind
     of adSemDefNameSym: n.diag.defNameSym
-    else: unreachable("all error cases must be covered")
+    else: unreachable("all error cases must be covered, got: " & $n.diag.kind)
```

Comment:
(no action required) If logging both a message plus an enum value in case of a failure is something that comes up often, then I think adding a dedicated `unreachable` overload would make sense.

Reply:
Smort!